packages/common/src/ui/layout/tables/columns/NameAndColorColumn.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { CellProps, Column } from 'react-table';
+import { TableCell } from '@material-ui/core';
+import { Color, ColorMenu, UnstyledIconButton } from '../../../components';
+import { Circle } from '../../../icons';
+import { Box } from '@material-ui/system';
+
+type Colorable = {
+  color: string;
+} & Record<string, unknown>;
+
+export const getNameAndColorColumn = <T extends Colorable>(
+  onChange: (row: T, color: Color) => void
+): Column<T> => ({
+  id: 'color',
+  align: 'left',
+
+  Cell: ({ row, value }: CellProps<T>) => {
+    const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+      setAnchorEl(event.currentTarget);
+    };
+
+    const handleClose = () => {
+      setAnchorEl(null);
+    };
+
+    return (
+      <TableCell
+        padding="none"
+        align="left"
+        sx={{
+          flexDirection: 'row',
+          borderBottom: 0,
+          alignItems: 'center',
+          display: 'flex',
+        }}
+      >
+        <ColorMenu
+          onClose={handleClose}
+          anchorEl={anchorEl}
+          onClick={color => {
+            handleClose();
+            onChange(row.original, color);
+          }}
```

Comment:
This is using the original data row we passed in. The `row` object has the values transformed using the `accessor` and also only contains the fields for the row (so color is technically missing in the `row` object, as the key for this row is `name`)